inst/private/extractblock.m

```diff
@@ -129,7 +129,16 @@
   switch wh
     case OCTCODE_INT
       assert(M == 1)
+      % Note: is it possible to avoid converting to double first?
```

Comment:
Unfortunately no, there are no ways to parse a string directly into a 64-bit integer at the moment, loss of precision seems inevitable at this point